lib/buffer.js

```diff
@@ -392,17 +392,25 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
-Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+Buffer.prototype.indexOf = function indexOf(val, byteOffset, encoding) {
   if (byteOffset > 0x7fffffff)
     byteOffset = 0x7fffffff;
   else if (byteOffset < -0x80000000)
     byteOffset = -0x80000000;
   byteOffset >>= 0;
 
-  if (typeof val === 'string')
-    return binding.indexOfString(this, val, byteOffset);
+  if (typeof val === 'string') {
+    if (encoding === undefined ||
+        encoding === 'ucs2' ||
+        encoding === 'utf16le' ||
+        encoding === 'utf8') {
+      return binding.indexOfString(this, val, byteOffset, encoding);
+    }
+    return binding.indexOfBuffer(
+        this, Buffer(val, encoding), byteOffset, encoding);
```

Comment:
I use the Buffer solution because the inputted string needs to be encoded before searched.
`String::WriteOneByte` is not usable in this context.

Both should yield same results:
Buffer('6264', 'hex').indexOf(Buffer('64', 'hex'))
Buffer('6264', 'hex').indexOf('64', 0, 'hex')